modules/cert-manager-enable-metrics.adoc

```diff
@@ -0,0 +1,85 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/cert-manager-operator-monitoring.adoc
+
+:_content-type: PROCEDURE
+[id="cert-manager-enable-metrics_{context}"]
+= Enabling monitoring for the {cert-manager-operator}
+
+You can enable monitoring and metrics collection for the {cert-manager-operator} by using a service monitor to performing custom metrics scraping.
+
+.Prerequisites
+
+* You have access to the cluster with `cluster-admin` privileges.
+* The {cert-manager-operator} is installed.
+// TODO: Any other things that need to have been done for Prometheus?
+
+.Procedure
+
+. TODO: Step on verifying Prometheus selectors.
++
+// TODO: From https://github.com/openshift/cert-manager-operator/blob/master/docs/OPERAND_METRICS.md, the first line is there but second line is empty (confirmed in a fresh cluster). Do we need to do something with this because of that, or is this what is expected?
+
+. Add the label to enable cluster monitoring by running the following command:
++
+[source,terminal]
+----
+$ oc label namespace cert-manager openshift.io/cluster-monitoring=true
+----
+
+. Enable monitoring for user-defined projects. See _Enabling monitoring for user-defined projects_ for instructions.
+
+. Create a service monitor.
+
+.. Create a YAML file that defines the `ServiceMonitor` object:
++
+.Example `service-monitor.yaml` file
+[source,yaml]
+----
+kind: ServiceMonitor
+metadata:
+  labels:
+    app: cert-manager
+    app.kubernetes.io/component: controller
+    app.kubernetes.io/instance: cert-manager
+    app.kubernetes.io/name: cert-manager
+    app.kubernetes.io/version: v1.10.2
+  name: cert-manager
+  namespace: cert-manager
+spec:
+  endpoints:
+  - interval: 30s
+    port: tcp-prometheus-servicemonitor
+    scheme: http
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: controller
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/name: cert-manager
+----
++
+// TODO: Make sure that this can be copied and applied as is
+
+.. Create the `ServiceMonitor` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f service-monitor.yaml
+----
++
+////
+TODO: Tested this and got the following error:
+[ahoffer@ahoffer march2] $ oc create -f service-monitor.yaml
+error: resource mapping not found for name: "cert-manager" namespace: "cert-manager" from "service-monitor.yaml": no matches for kind "ServiceMonitor" in version ""
+ensure CRDs are installed first
+
+apiVersion: monitoring.coreos.com/v1 was missing from the example YAML in the
+////
++
+// TODO: this command in https://github.com/openshift/cert-manager-operator/blob/master/docs/OPERAND_METRICS.md passed in "-n certmanager", but that isn't necessary since it's defined in the resource, right?
```

Comment:
@thejasn @swghosh Can you confirm whether the "-n certmanager" is required on "oc create -f service-monitor.yaml"? It was in the docs you had, but I didn't think it was needed since it was specified in the YAML file.

Reply:
You've removed metrics for the GA docs correct?

Reply:
@thejasn yes that's correct, though this is still something that I'll need to know for when I'm ready to add them in.

Reply:
Yes the explicit namespace isn't required in this case.